fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/PostBreakBlockCallback.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.player;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Callback before a block is broken.
+ * Only called on the server, however updates are synced with the client.
+ *
+ * <p>Upon return:
+ * <ul><li>SUCCESS/PASS/CONSUME continues the default code for breaking the block
+ * <li>FAIL cancels the block breaking action</ul>
+ */
+public interface PostBreakBlockCallback {
```

Comment:
Named post but the event is called before? Something is wrong here.

Reply:
Imo just make a `BlockBreakEvents` class and move the individual callback interfaces to subclasses there with simple names like `Pre` `Post`